Kitodo/src/main/java/org/kitodo/services/data/BatchService.java

```diff
@@ -281,10 +281,12 @@ public void addAllObjectsToIndex() throws CustomResponseException, InterruptedEx
     @Override
     public BatchDTO convertJSONObjectToDTO(JSONObject jsonObject, boolean related) throws DataException {
         BatchDTO batchDTO = new BatchDTO();
-        batchDTO.setTitle(getStringPropertyForDTO(jsonObject, "title"));
-        batchDTO.setType(getStringPropertyForDTO(jsonObject, "type"));
+        batchDTO.setId(getIdFromJSONObject(jsonObject));
+        JSONObject batchObject = getSource(jsonObject);
```

Comment:
why did you move the "getSource" out of the methods?

Reply:
I wanted to avoid this one level querying _source for every property. Probably it doesn't give much increase of the speed for one object, but for 1000 or 10000 it can already make a difference.